zwave_js_server/model/value.py

```diff
@@ -42,13 +42,35 @@ class ValueDataType(TypedDict, total=False):
     ccVersion: int
 
 
-def get_value_id(node: "Node", event_data: ValueDataType) -> str:
+def _get_value_id_from_dict(node: "Node", val: ValueDataType) -> str:
+    """Return ID of value from ValueDataType dict."""
+    return get_value_id(
+        node,
+        val["commandClass"],
+        val["property"],
+        val.get("endpoint"),
+        val.get("propertyKey"),
+        val.get("propertyKeyName"),
+    )
+
+
+def get_value_id(
+    node: "Node",
+    command_class: int,
+    property_: Union[str, int],
+    endpoint: Optional[Union[str, int]] = None,
```

Comment:
According to the zwave-js docs endpoint is an optional number. Is that correct?
```suggestion
    endpoint: Optional[int]] = None,
```

Reply:
yes, this was a `mypy` error I was resolving but I can fix it differently